### PR TITLE
perf(install): remove --debug, remove apt-get update/upgrade

### LINKED ISSUE #63: Install is too long
Here is an install log from the debug interface.
The entire install takes 53 minutes.

The low hanging fruits:
- pre-installing debian packages (currently taking 12 minutes)
- Installing the Wifi hotspot takes 9 minutes. This should be investigated.

A subsidiary question: should we install Roundcube by default (6 minutes) ? I would say no.

```
2019-09-21 12:50: [INFO] ===== Start HyperCube Service =====
2019-09-21 12:50: [INFO] Looking for HyperCube file
2019-09-21 12:50: [INFO] Heating up ...
2019-09-21 12:50: [INFO] Install some dependencies...
2019-09-21 12:53: [INFO] Detecting USB sticks...
2019-09-21 12:53: [INFO] HyperCube file found
2019-09-21 12:53: [INFO] Loading JSON
2019-09-21 12:53: [INFO] Extracting settings for Unix
2019-09-21 12:53: [INFO] Extracting settings for YunoHost
2019-09-21 12:53: [INFO] Extracting settings for Wifi Hotspot
2019-09-21 12:53: [INFO] Extracting settings for VPN Client (logging)
2019-09-21 12:53: [INFO] Extracting .cube file for VPN Client
2019-09-21 12:53: [INFO] Updating Debian root password
2019-09-21 12:53: [INFO] Changing hostname
2019-09-21 12:53: [INFO] Updating hosts file
2019-09-21 12:53: [INFO] Setting locales
2019-09-21 12:53: [INFO] Upgrading Debian/YunoHost...
2019-09-21 13:05: [INFO] Check online DynDNS domains list
2019-09-21 13:05: [INFO] Doing YunoHost post-installation...
2019-09-21 13:18: [INFO] Removing DynDNS cron
2019-09-21 13:18: [INFO] Fetching YunoHost apps list for labriqueinternet
2019-09-21 13:18: [INFO] Creating first user
2019-09-21 13:18: [INFO] Installing VPN Client...
2019-09-21 13:21: [INFO] Installing Wifi Hotspot...
2019-09-21 13:32: [INFO] Installing Roundcube Webmail...
2019-09-21 13:38: [INFO] Configuring VPN Client...
2019-09-21 13:40: [INFO] Configuring Wifi Hotspot...
2019-09-21 13:40: [INFO] Rebooting...
2019-09-21 13:43: [INFO] ===== Start HyperCube Service =====
2019-09-21 13:43: [INFO] Starting second step
2019-09-21 13:43: [INFO] Wifi device correctly detected after rebooting
2019-09-21 13:43: [INFO] Finished!
2019-09-21 13:43: [INFO] No USB stick detected for log copying
2019-09-21 13:43: [INFO] 4 hours (without reboot) before disabling this interface
2019-09-21 13:43: [INFO] Please, save this page with Ctrl+S
```

### PR DESCRIPTION
**Pull request by HugoPoi**
Fix #63 
* Usb deps are now pre-install inside the base image [see here](https://github.com/keomabrun/arm-images/pull/1/commits/a3e7dbd96aee284b92ed2613603f9b21aeadf54d)
* Remove all the yunohost `--debug` giving a huge penalty on exec time
* `logfilter` function need a rewrite if we want to used it again (I think we need to trash it)
* Remove the apt upgrade, as we can CI/CD the image

### CODE REVIEW

File: script/hypercube/hypercube.sh [hunk 1]
```diff
@@ -352,14 +345,14 @@ function deb_setlocales_and_tz() {
...
+  yunohost tools postinstall -d "${settings[yunohost,domain]}" -p "${settings[yunohost,password]}" &>> $log_file
 }
 
 function ynh_addappslist() {
   logfile ${FUNCNAME[0]}
 
-  yunohost app fetchlist -n community -u https://app.yunohost.org/community.json --debug &>> $log_file
-  yunohost app fetchlist --debug &>> $log_file
+  yunohost app fetchlist -n community -u https://app.yunohost.org/community.json &>> $log_file
+  yunohost app fetchlist &>> $log_file
```
Comment (zamentur) [inline script/hypercube/hypercube.sh]: We shouldn't do this now, because vpnclient_ynh and hotsport are directly manage in the official list of yunohost.

**Review by zamentur — COMMENTED** [review]
About logfilter, it's used to avoid to display password in the debug page of the internet cube.
But, now yunohost include a mechanism to hide password too. I don't remember if output and logfile are concerned.

I think debug page could simply display yunohost history instead of creating redundant logfile.

**Comment by HugoPoi** [comment]
I think should remove the `logfilter` function, useless because Yunohost implement already a hiding mechanism as you suggested and I already redirected all yunohost output to our log files

**Comment by HugoPoi** [comment]
Image https://github.com/HugoPoi/build.labriqueinter.net/releases/tag/v3.6.5.3-RC1

**Comment by HugoPoi** [comment]
- [ ] Open issue for adding an option for `apt upgrade`
- [ ] Open issue for adding option for `--debug` (@pitchum insist to have it just for him)

**Comment by SohKa** [comment]
Has the `test` dir been added on purpose ? I would suppose not, so I think it should be excluded with a gitignore file.

File: test/lime2.ovpn [hunk 2]
```diff
@@ -0,0 +1,110 @@
+
+client
+nobind
+dev tun
+remote-cert-tls server
+
+remote 192.168.1.141 1194 udp
+
+<key>
+-----BEGIN PRIVATE KEY-----
```
Comment (alexAubin) [inline test/lime2.ovpn]: yo dawg are you really versionning a working private key ? :s

File: script/hypercube/hypercube.sh [hunk 3]
```diff
@@ -585,7 +571,7 @@ function monitoring_yunohost() {
 
     date >> $tmplog
     echo -e "\n" >> $tmplog
-    yunohost tools diagnosis --debug &>> $tmplog
+    yunohost tools diagnosis &>> $tmplog
```
Comment (alexAubin) [inline script/hypercube/hypercube.sh:578]: N.B. : this command gonna get removed in 3.8 ... personally I never found this command so useful except to display yunohost version, which can also be obtained from yunohost --version :/

File: test/lime2.ovpn (hunk 2 above)
Comment (HugoPoi) [reply test/lime2.ovpn]: local testing key (one time used key for test), but yes I need to remove-it from here and have a script for generate the hypercube for testing

File: script/hypercube/hypercube.sh (hunk 3 above)
Comment (HugoPoi) [reply script/hypercube/hypercube.sh:578]: Ok thanks for pointing this, I will remove it for preparing the v3.8

**Comment by HugoPoi** [comment]
@SohKa Yes it's on purpose, the idea it's to have a way of testing the generated image

File: script/hypercube/hypercube.sh [hunk 4]
```diff
@@ -196,13 +196,6 @@ function find_hypercubefile() {
   # using systemd services but meh I'm lazy and not very expert
   sleep 10
 
-  info "Install some dependencies..."
-  rm -rf /var/lib/apt/lists/*
-  apt-get clean &>> $log_file
-  apt-get update &>> $log_file
-
-  apt-get install -o Dpkg::Options::='--force-confold' -y --force-yes file udisks2 udiskie ntfs-3g jq  &>> $log_file || true
```
Comment (HugoPoi) [inline script/hypercube/hypercube.sh:204]: * Removing this line depends on the this patch https://github.com/keomabrun/arm-images/commit/a3e7dbd96aee284b92ed2613603f9b21aeadf54d

So we can't merge this right now

File: script/hypercube/hypercube.sh (hunk 3 above)
Comment (HugoPoi) [reply script/hypercube/hypercube.sh:578]: If someone know the utility of `monitoring_yunohost()` because I think we can remove it ?

**Pull request merged**